CONTRIBUTING.rst

```diff
@@ -155,18 +114,101 @@ Contributing to the documentation
 can build the documentation with:
 
 .. code-block:: bash
 
     tox -e docs
 
-You can visualise the local documentation with your favorite browser (here
+You can visualize the local documentation with your favorite browser (here
 Mozilla Firefox is used)
 
 .. code-block:: bash
 
     firefox docs/build/html/index.html
 
 .. _`sphinx` : https://www.sphinx-doc.org/en/master/
+
+Python doc strings
+~~~~~~~~~~~~~~~~~~
+
+Our docstring format follows the `sphinx format`_ and a typical function doc string
+looks like the following.
+
+.. code-block:: python
+
+    def func(value_1: float, value_2: int) -> float:
+      r"""A one line summary sentence of the function.
+
+      Extensive multi-line summary of what is going in. Use single 
+      backticks for parameters of the function like `width` and two ticks for
+      values ``67``. You can link to classes :py:class:`equistore.Labels`. This
+      also works for other classes and functions like :py:obj:`True`.
+
+      Inline Math is also possible with :math:`\mathsf{R}`. Or as a math block.
+
+      .. math::
+
+          \mathbf{x}' = \mathsf{R}\mathbf{x}
+
+
+      :param value_1:
+          The first parameter of the function, a :py:class:`float`.
+      :param value_2:
+          The second parameter of the function, an :py:class:`int`.
+
+      :returns result:
+          The result of the calculation, a :py:class:`float`.
+
+      :raises ValueError:
+          if the value is not a float
+
+      Examples
+      --------
+
+      This is an example
+
+      >>> from equistore import func
+      >>> a = np.array(
+      ...    [1, 2, 3, 4]
+      ... )
+      >>> func(1, 1)
+      42
+    """
+    ...
+    return result
+
+Guidelines for writing Python doc strings
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* Use Python typing in the function arguments, indicate return types.
+
+* Start the description after each ``:param:`` or ``:return:`` in a new line and add an
+  empty line between the parameter and return block.
+
+* Emphasize function and class parameters with a single backtick i.e ```param``` and
+  general variables should be double backticked . i.e. ````my_variable````
+
+* If you include any maths, make the string a
+  `raw string`_ by prefixing with ``r``, e.g.,
+
+  .. code-block:: python
+
+    r"""Some math like :math:`\nu^2 / \rho` with backslashes."""
+
+  Otherwise the ``\n`` and ``\r`` will be rendered as ASCII escape sequences that break
+  lines without you noticing it or you will get either one of the following two
+  errors message
+
+  1. `Explicit markup ends without a blank line; unexpected unindent`
+  2. `Inline interpreted text or phrase reference start-string without end string`
+
+* The examples are tested with `doctest`_. Therefore, please make sure that they are
+  complete and functioning (with all required imports). As given in the example above 
+  use the ``>>>`` syntax for inputs (followed by ``...`` for multiline inputs) and no
```

Comment:
I'd move the `...` example in the example doc string above to here, as it has nothing to do with the example `func()`